python/tests/test_array.py

```diff
@@ -657,6 +657,58 @@ def test_array_iteration(self):
         self.assertEqual(y.tolist(), [3.0, 4.0])
         self.assertEqual(z.tolist(), [5.0, 6.0])
 
+    def test_array_pickle(self):
+        dtypes = [
+            mx.int8,
+            mx.int16,
+            mx.int32,
+            mx.int64,
+            mx.uint8,
+            mx.uint16,
+            mx.uint32,
+            mx.uint64,
+            # mx.float16,
+            mx.float32,
+            # mx.bfloat16,
```

Comment:
Why not float16 and bfloat16?

Reply:
My bad, I forgot to uncomment after closing #702 . Fixing soon